src/flatten.rs

```diff
@@ -53,32 +65,22 @@ impl Visitor for ReferencesCollector {
         }
     }
 
-    fn visit_user_defined_type_name(&mut self, type_name: &UserDefinedTypeName) {
-        self.process_referenced_declaration(type_name.referenced_declaration, &type_name.src);
-    }
-
-    fn visit_member_access(&mut self, member_access: &MemberAccess) {
-        if let Some(id) = member_access.referenced_declaration {
-            self.process_referenced_declaration(id, &member_access.src);
-        }
-    }
-
     fn visit_identifier_path(&mut self, path: &IdentifierPath) {
         self.process_referenced_declaration(path.referenced_declaration, &path.src);
     }
-}
 
-/// Visitor exploring AST and collecting all references to any declarations found in
-/// `UserDefinedTypeName` nodes
-struct UserDefinedTypeNamesCollector {
-    path: PathBuf,
-    references: HashMap<isize, HashSet<ItemLocation>>,
-}
-
-impl Visitor for UserDefinedTypeNamesCollector {
-    fn visit_user_defined_type_name(&mut self, type_name: &UserDefinedTypeName) {
-        if let Some(loc) = ItemLocation::try_from_source_loc(&type_name.src, self.path.clone()) {
-            self.references.entry(type_name.referenced_declaration).or_default().insert(loc);
+    fn visit_member_access(&mut self, access: &MemberAccess) {
+        if let Some(referenced_declaration) = access.referenced_declaration {
+            let length = access.member_name.len();
+            // Accessed member name is in the last name.len() symbols of the expression.
+            let start = access.src.start.unwrap() + access.src.length.unwrap() - length;
+            let end = start + length;
```

Comment:
this should be safe, right?

Reply:
It's safe for latest versions, however thinking of it now it might be better to make it a bit safer as I'm not sure what might be happening on older versions

will push a fix